### DISCUSSION TITLE
Auth module not respecting dynamic router.base (Nuxt 2.15+)

### DISCUSSION THREAD

**Discussion by MrNaif2018** [nuxt-community/auth-module#1625] [Support Requests]
### Version
module: 5.0.0-1613647907.37b1156 (same with 4.x.x. series)
nuxt: 2.15.1

#### [mode](https://nuxtjs.org/api/configuration-mode): 
 - [x] universal

### Nuxt configuration
```
auth: {
    localStorage: false,
    cookie: {
      options: {
        expires: 7,
      },
    },
    strategies: {
      local: {
        endpoints: {
          login: {
            url: "/token",
            method: "post",
            propertyName: "access_token",
          },
          logout: false,
          user: { url: "/users/me", method: "get", propertyName: false },
        },
      },
    },
    plugins: ["~/plugins/auth.js"],
  },
```

## plugins/auth.js

```
export default ({ $axios, $auth }) => {
  $axios.onError((error) => {
    if (
      error.response &&
      error.response.status === 401 &&
      !error.config.url.endsWith("/token")
    ) {
      $auth.logout()
    }
  })
}
```

## Reproduction
Codesandbox seems to reject process.env, but basically, nuxt.config.js must have:

```
router: {
    base: process.env.ROOTPATH || "/",
  },
```

Then, run `yarn build`.
`yarn start` - all works as expected
`ROOTPATH=/admin yarn start` - by opening `/admin`, it redirects to `/login` and not to `/admin/login`.
Opening correct page manually all works.

## Checklist
* [x]  I have tested with the latest Nuxt version and the issue still occurs
* [x]  I have tested with the latest module version and the issue still occurs
* [x]  I have searched the issue tracker and this issue hasn't been reported yet

### What is expected?

A correct redirect to `/admin/login` to be done

### What is actually happening?
Redirect to `/login`, ignoring router.base.
Note: it works if router.base is set statically (not via env var).
So maybe the module is not yet ready for nuxt/nuxt.js#8520
![изображение](https://user-images.githubusercontent.com/39452697/108753522-e9cb0b00-7555-11eb-9edb-3c7088fec59a.png)
This second one is what I get after opening localhost:3000/admin:
![изображение](https://user-images.githubusercontent.com/39452697/108753569-f64f6380-7555-11eb-998f-1011d32a4734.png)

If possible, could the fix be backported to the 4.x.x series?

**Comment by MrNaif2018** [comment]
Found a workaround (in auth plugin):
```
$auth.onRedirect((to, from) => {
    if (!process.server) return
    if (to.startsWith(decodeURI($config.app.basePath))) return to
    return urlJoin($config.app.basePath, to)
  })
```

**Comment by sogehige** [comment]
> Found a workaround (in auth plugin):
> 
> ```
> $auth.onRedirect((to, from) => {
>     if (!process.server) return
>     if (to.startsWith(decodeURI($config.app.basePath))) return to
>     return urlJoin($config.app.basePath, to)
>   })
> ```

Doesn't seem working for `home` redirect. Any ideas?

**Comment by MrNaif2018** [comment]
> Doesn't seem working for `home` redirect. Any ideas?

Unfortunately no, this has been working fine for me for a while.

@JoaoPedroAS51 Sorry for ping, could you take a look at this bug? It has been there for a while in nuxt/auth, both v4 and v5, and looks like it hasn't been resolved yet.
